lib/array_equals.rb

```diff
@@ -1,5 +1,12 @@
-# Determines if the two input arrays have the same count of elements
-# and the same integer values in the same exact order
+
 def array_equals(array1, array2)
-  raise NotImplementedError
+  return true if array1 == nil and array2 == nil
+  return false if !array1 or !array2
```

Comment:
These two lines correctly handle all the cases where one array is nil, but there are two readability/maintainability problems. 

First, you use (array1 == nil) on one line and then (!array1) on the next line to mean the same thing. Pick one and be consistent, at least within the method. (!array1) is fine for a small codebase where there are no team members who object to terse code. So, basically, not a real-world team. (array1 == nil) is much clearer.

Second, these two lines look like two independent checks for error conditions, but in reality the second line depends on the first. It's the cleanest way to write the code, but makes it easy for a future maintenance programmer to muck it up. Maybe add a comment.